File: src/main/java/cc/moecraft/logger/AnsiColor.java
package cc.moecraft.logger;

import static cc.moecraft.logger.AnsiConsants.PREFIX;
import static cc.moecraft.logger.AnsiConsants.SUFFIX;

/**
 * 此类由 Hykilpikonna 在 2018/05/04 创建!
 * Created by Hykilpikonna on 2018/05/04!
 * Github: https://github.com/hykilpikonna
 * Meow!
 *
 * @author Hykilpikonna
 */
public enum AnsiColor
{
    RESET(0),

    BLACK(30),
    RED(31),
    GREEN(32),
    YELLOW(33),
    BLUE(34),
    PURPLE(35),
    CYAN(36),
    WHITE(37);

    int code;

    AnsiColor(int code)
    {
        this.code = code;
    }

    @Override
    public String toString()
    {
        return PREFIX + code + SUFFIX;
    }

    public String get()
    {
        return toString();
    }

    public String getBright()
    {
        return PREFIX + code + ";1" + SUFFIX;
    }

    public String getBackground()
    {
        return PREFIX + (code + 10) + SUFFIX;
    }
}

File: src/main/kotlin/org/hydev/logger/format/AnsiColor.kt
package org.hydev.logger.format

import org.hydev.logger.format.AnsiConstants.ESC_PREFIX
import org.hydev.logger.format.AnsiConstants.SUFFIX

@Suppress("ConvertToStringTemplate")
enum class AnsiColor(val code: Int, vararg val placeholders: Char)
{
    RESET(0, 'r'),
    BLACK(30, '0', '8'),
    RED(31, '4', 'c'),
    GREEN(32, '2', 'a'),
    YELLOW(33, '6', 'e'),
    BLUE(34, '1', '9'),
    PURPLE(35, '5', 'd'),
    CYAN(36, '3', 'b'),
    WHITE(37, '7', 'f');

    override fun toString() = value

    val value = ESC_PREFIX + code + SUFFIX
    val bright = ESC_PREFIX + code + ";1" + SUFFIX
    val background = ESC_PREFIX + (code + 10) + SUFFIX
    val brightBg = ESC_PREFIX + (code + 70) + SUFFIX
}
